tree/ntuple/v7/src/RField.cxx

```diff
@@ -1940,7 +1941,8 @@ void ROOT::Experimental::RRVecField::ReadGlobalImpl(NTupleSize_t globalIndex, vo
       }
 
       // TODO Increment capacity by a factor rather than just enough to fit the elements.
-      free(*beginPtr);
+      if (owns)
+         free(*beginPtr);
```

Comment:
if I understand correctly the changes below, aren't RVecs read with RNTuple _always_ non-owning now?

Reply:
or: when are they not owning?

Reply:
ah no sorry, we set the capacity to -1 in GenerateValue but then we reset it to a positive value in here.

so now I'm confused as to when RVecs read by RNTuple are owning and when they are not :grimacing:

Reply:
At the moment, they are not owning when used with `BulkRead`. In the scalar `Read`, RNTuple allocates the memory for the vector contents and updates the capacity pointer accordingly.

Reply:
Perhaps it would be useful to add a code comment on this :slightly_smiling_face:.